pyiron_base/job/worker.py

```diff
@@ -207,7 +221,8 @@ def run_static_with_database(self):
                         for pp, p, job_id in path_lst
                     ]
                     active_job_ids += [j[1] for j in job_lst]
-                    pool.map_async(worker_function, job_lst)
+                    result = pool.map_async(worker_function, job_lst)
+                    res_lst.append([result, len(job_lst)])
```

Comment:
Nothing happens with `res_lst` yet, right?  Maybe we can directly await on it before the worker shuts down.  Otherwise I've had situations happen, where you set the job status to `collect`, but it was still executing jobs, which then get silently dropped.

Reply:
Currently it is primarily used to to fill the queue:
```
sum([i for r, i in res_lst if not r.ready()]) < number_tasks * self.input.queue_limit_factor
```
I personally use ability to set to `collect` to cancel the worker if something failed.

Reply:
I like the suggestion and added it.